--- libmemcached-1.0.16.orig/m4/libtool.m4	2013-01-24 09:09:18.000000000 -0500
+++ libmemcached-1.0.16/m4/libtool.m4	2015-10-27 14:49:15.456423455 -0400
@@ -1345,7 +1345,10 @@
 	  x86_64-*linux*)
 	    LD="${LD-ld} -m elf_x86_64"
 	    ;;
-	  ppc*-*linux*|powerpc*-*linux*)
+          ppc64le*-*linux*|powerpc64le*-*linux*)
+            LD="${LD-ld} -m elf64lppc"
+            ;;
+          ppc*-*linux*|powerpc*-*linux*)
 	    LD="${LD-ld} -m elf64ppc"
 	    ;;
 	  s390*-*linux*|s390*-*tpf*)
