--- json-c-json-c-0.12-20140410.orig/configure	2014-04-10 22:35:45.000000000 -0400
+++ json-c-json-c-0.12-20140410/configure	2015-10-15 14:32:51.511578406 -0400
@@ -7190,6 +7190,9 @@
 	  x86_64-*linux*)
 	    LD="${LD-ld} -m elf_x86_64"
 	    ;;
+         ppc64le*-*linux*|powerpc64le*-*linux*)
+           LD="${LD-ld} -m elf64lppc"
+           ;;
 	  ppc*-*linux*|powerpc*-*linux*)
 	    LD="${LD-ld} -m elf64ppc"
 	    ;;
