--- sds-1.0.0.orig/sds.h	2014-02-06 11:10:16.000000000 -0500
+++ sds-1.0.0/sds.h	2015-10-13 13:00:25.119414916 -0400
@@ -36,6 +36,10 @@
 #include <sys/types.h>
 #include <stdarg.h>
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 typedef char *sds;
 
 struct sdshdr {
@@ -45,12 +49,12 @@
 };
 
 static inline size_t sdslen(const sds s) {
-    struct sdshdr *sh = (void*)(s-sizeof *sh);
+    struct sdshdr *sh = (struct sdshdr *)(s-sizeof *sh);
     return sh->len;
 }
 
 static inline size_t sdsavail(const sds s) {
-    struct sdshdr *sh = (void*)(s-sizeof *sh);
+    struct sdshdr *sh = (struct sdshdr *)(s-sizeof *sh);
     return sh->free;
 }
 
@@ -98,4 +102,8 @@
 sds sdsRemoveFreeSpace(sds s);
 size_t sdsAllocSize(sds s);
 
+#ifdef __cplusplus
+}
+#endif
+
 #endif
