--- cpp-hiredis-cluster-master.orig/include/clusterexception.h	2015-10-01 12:40:19.742455136 -0400
+++ cpp-hiredis-cluster-master/include/clusterexception.h	2015-10-02 11:21:41.153423684 -0400
@@ -31,6 +31,8 @@
 
 #include <stdexcept>
 #include <string.h>
+#include <stdio.h>
+#include <errno.h>
 
 namespace RedisCluster
 {
@@ -118,4 +120,4 @@
     };
 }
 
-#endif // defined(__libredisCluster__clusterexception__)
\ No newline at end of file
+#endif // defined(__libredisCluster__clusterexception__)
