--- cpp-hiredis-cluster-master.orig/include/container.h	2015-10-01 12:40:19.752487494 -0400
+++ cpp-hiredis-cluster-master/include/container.h	2015-10-02 11:20:22.192424623 -0400
@@ -91,7 +91,7 @@
             {
             }
             
-            typename RCluster::HostConnection conn( key, connect_( host.c_str(), std::stoi(port), data_ ) );
+            typename RCluster::HostConnection conn( key, connect_( host.c_str(), atoi(port.c_str()), data_ ) );
             if( conn.second != NULL && conn.second->err == 0 )
             {
                 connections_.insert( conn );
