--- curl-7.29.0.orig/config.guess	2013-01-20 17:03:06.000000000 -0500
+++ curl-7.29.0/config.guess	2015-10-20 11:14:20.036798538 -0400
@@ -4,7 +4,7 @@
 #   2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010,
 #   2011, 2012 Free Software Foundation, Inc.
 
-timestamp='2012-02-10'
+timestamp='2015-10-20'
 
 # This file is free software; you can redistribute it and/or modify it
 # under the terms of the GNU General Public License as published by
@@ -969,6 +969,9 @@
 	  *)    echo hppa-unknown-linux-gnu ;;
 	esac
 	exit ;;
+    ppc64le:Linux:*:*)
+       echo powerpc64le-unknown-linux-gnu
+       exit ;;
     ppc64:Linux:*:*)
 	echo powerpc64-unknown-linux-gnu
 	exit ;;
